<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>849</width>
    <height>561</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>LED Light</string>
  </property>
  <property name="windowIcon">
   <iconset resource="default.qrc">
    <normaloff>:/icon.svg</normaloff>:/icon.svg</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_Images">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>221</width>
      <height>531</height>
     </rect>
    </property>
    <property name="title">
     <string>Images</string>
    </property>
    <widget class="QListWidget" name="listWidget">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>181</width>
       <height>421</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::NoFocus</enum>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOn</enum>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::NoSelection</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_Del">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>50</y>
       <width>31</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_Browse">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>20</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>+</string>
     </property>
     <property name="shortcut">
      <string>Ctrl+O</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_Height">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Height:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_Height">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>50</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>800</number>
     </property>
     <property name="value">
      <number>8</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_Width">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Width:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_Width">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>20</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>800</number>
     </property>
     <property name="value">
      <number>8</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_Clear">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>50</y>
       <width>31</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>C</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_Output">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>450</y>
      <width>571</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Output</string>
    </property>
    <widget class="QSpinBox" name="spinBox_One">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>20</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>104</number>
     </property>
     <property name="value">
      <number>15</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_Zero">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>20</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>104</number>
     </property>
     <property name="value">
      <number>2</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_One">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>20</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>One:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_Clock_Divider">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Clock divider:</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBox_CH">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>51</y>
       <width>121</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Generate C/H files</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_Pulse">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>20</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Pulse:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_Clock_Divider">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>20</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>65535</number>
     </property>
     <property name="value">
      <number>4</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_Pulse">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>20</y>
       <width>41</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>104</number>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBox_Bin">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>51</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Generate binary file</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_Zero">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>20</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Zero:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_Convert">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>470</x>
       <y>51</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Convert</string>
     </property>
     <property name="shortcut">
      <string>Ctrl+S</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_Stop_Pulses">
     <property name="geometry">
      <rect>
       <x>510</x>
       <y>20</y>
       <width>41</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>65535</number>
     </property>
     <property name="value">
      <number>40</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_Stop_Pulses">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>20</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop pulses:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_Byte_Order">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>50</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>GRB</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>RGB</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_Byte_Order">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Byte order:</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_Mask">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>10</y>
      <width>571</width>
      <height>431</height>
     </rect>
    </property>
    <property name="title">
     <string>Mask</string>
    </property>
    <widget class="QTableWidget" name="tableWidget">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>531</width>
       <height>361</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::NoFocus</enum>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="tabKeyNavigation">
      <bool>false</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <attribute name="horizontalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="horizontalHeaderMinimumSectionSize">
      <number>4</number>
     </attribute>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>16</number>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="verticalHeaderMinimumSectionSize">
      <number>4</number>
     </attribute>
     <attribute name="verticalHeaderDefaultSectionSize">
      <number>16</number>
     </attribute>
    </widget>
    <widget class="QSpinBox" name="spinBox_Point_Size">
     <property name="geometry">
      <rect>
       <x>510</x>
       <y>390</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>4</number>
     </property>
     <property name="maximum">
      <number>80</number>
     </property>
     <property name="value">
      <number>16</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_Point_Size">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>390</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Point size:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_Scan_Direction">
     <property name="geometry">
      <rect>
       <x>108</x>
       <y>390</y>
       <width>211</width>
       <height>22</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Left to Right / Top to Bottom / Zig-zag</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Right to Left / Top to Bottom / Zig-zag</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Left to Right / Bottom to Top / Zig-zag</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Right to Left / Bottom to Top / Zig-zag</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Top to Bottom / Left to Right / Zig-zag</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Bottom to Top / Left to Right / Zig-zag</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Top to Bottom / Right to Left / Zig-zag</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Bottom to Top / Right to Left / Zig-zag</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Left to Right / Top to Bottom / Lines</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Right to Left / Top to Bottom / Lines</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Left to Right / Bottom to Top / Lines</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Right to Left / Bottom to Top / Lines</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Top to Bottom / Left to Right / Lines</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Bottom to Top / Left to Right / Lines</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Top to Bottom / Right to Left / Lines</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Bottom to Top / Right to Left / Lines</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_Scan_Direction">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>390</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Scan direction:</string>
     </property>
    </widget>
   </widget>
  </widget>
  <action name="actionBrowseAction">
   <property name="text">
    <string>BrowseAction</string>
   </property>
  </action>
  <action name="actionConvertAction">
   <property name="text">
    <string>ConvertAction</string>
   </property>
  </action>
  <action name="actionTableCellChangedAction">
   <property name="text">
    <string>TableCellChangedAction</string>
   </property>
   <property name="toolTip">
    <string>TableCellChangedAction</string>
   </property>
  </action>
  <action name="actionTableResizeAction">
   <property name="text">
    <string>TableResizeAction</string>
   </property>
  </action>
  <action name="actionDelAction">
   <property name="text">
    <string>DelAction</string>
   </property>
  </action>
  <action name="actionPulseChangedAction">
   <property name="text">
    <string>PulseChangedAction</string>
   </property>
  </action>
  <action name="actionPointSizeChangedAction">
   <property name="text">
    <string>PointSizeChangedAction</string>
   </property>
  </action>
  <action name="actionCheckBoxGenerateAction">
   <property name="text">
    <string>CheckBoxGenerateAction</string>
   </property>
  </action>
  <action name="actionClockZeroOneStopSpinBoxChangedAction">
   <property name="text">
    <string>ClockZeroOneStopSpinBoxChangedAction</string>
   </property>
  </action>
  <action name="actionClearAction">
   <property name="text">
    <string>ClearAction</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>spinBox_Width</tabstop>
  <tabstop>spinBox_Height</tabstop>
  <tabstop>pushButton_Browse</tabstop>
  <tabstop>pushButton_Del</tabstop>
  <tabstop>pushButton_Clear</tabstop>
  <tabstop>comboBox_Scan_Direction</tabstop>
  <tabstop>spinBox_Point_Size</tabstop>
  <tabstop>spinBox_Clock_Divider</tabstop>
  <tabstop>spinBox_Pulse</tabstop>
  <tabstop>spinBox_Zero</tabstop>
  <tabstop>spinBox_One</tabstop>
  <tabstop>spinBox_Stop_Pulses</tabstop>
  <tabstop>comboBox_Byte_Order</tabstop>
  <tabstop>checkBox_CH</tabstop>
  <tabstop>checkBox_Bin</tabstop>
  <tabstop>pushButton_Convert</tabstop>
 </tabstops>
 <resources>
  <include location="default.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_Browse</sender>
   <signal>clicked()</signal>
   <receiver>actionBrowseAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_Convert</sender>
   <signal>clicked()</signal>
   <receiver>actionConvertAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_Height</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionTableResizeAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>405</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_Width</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionTableResizeAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_Del</sender>
   <signal>clicked()</signal>
   <receiver>actionDelAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>185</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_Pulse</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionPulseChangedAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>500</x>
     <y>480</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableWidget</sender>
   <signal>currentCellChanged(int,int,int,int)</signal>
   <receiver>actionTableCellChangedAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>310</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_Point_Size</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionPointSizeChangedAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>790</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_Bin</sender>
   <signal>stateChanged(int)</signal>
   <receiver>actionCheckBoxGenerateAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>465</x>
     <y>510</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_CH</sender>
   <signal>stateChanged(int)</signal>
   <receiver>actionCheckBoxGenerateAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>340</x>
     <y>510</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_Clock_Divider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionClockZeroOneStopSpinBoxChangedAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>480</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_Zero</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionClockZeroOneStopSpinBoxChangedAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>570</x>
     <y>480</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_One</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionClockZeroOneStopSpinBoxChangedAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>660</x>
     <y>480</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_Stop_Pulses</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionClockZeroOneStopSpinBoxChangedAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>790</x>
     <y>480</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_Scan_Direction</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>actionClockZeroOneStopSpinBoxChangedAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>458</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_Clear</sender>
   <signal>clicked()</signal>
   <receiver>actionClearAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>205</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_Byte_Order</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>actionClockZeroOneStopSpinBoxChangedAction</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>510</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
